### ISSUE TITLE
Not Compatible with OpenTracing Tracer Interface

### ISSUE THREAD

**Issue by kunicmarko20** [DataDog/dd-trace-php#476]
Given this code:

```
use OpenTracing\GlobalTracer;
use OpenTracing\StartSpanOptions;
use OpenTracing\Tags;

      $scope = GlobalTracer::get()->startActiveSpan(
            'my_name',
            StartSpanOptions::create(
                [
                    'tags' => [
                        Tags\SPAN_KIND => Tags\SPAN_KIND_MESSAGE_BUS_PRODUCER,
                        'message_id' => 'some id',
                        //...
                    ],
                ]
            )
        );
```

https://github.com/DataDog/dd-trace-php/blob/9670268570b671bb0ba5cb6633564571515d0fe7/src/DDTrace/Tracer.php#L236

On this line you are checking for `DDTrace\StartSpanOptions` and we are passing `OpenTracing\StartSpanOptions`.

The interface https://github.com/opentracing/opentracing-php/blob/0c421ae732bbcf4765088ae3f212c20541b4c13f/src/OpenTracing/Tracer.php#L47
explains we can pass `OpenTracing\StartSpanOptions` and your integration with OpenTracing https://github.com/DataDog/dd-trace-php/blob/9670268570b671bb0ba5cb6633564571515d0fe7/src/DDTrace/OpenTracer/Tracer.php#L59
Just delegates to https://github.com/DataDog/dd-trace-php/blob/9670268570b671bb0ba5cb6633564571515d0fe7/src/DDTrace/Tracer.php#L234 without converting to `array` or `DDTrace\StartSpanOptions` and then it fails.

**Comment by pawelchcki** [comment]
Thanks for reporting! Looks like our implementation might have diverged too much over last few months and we need fix that and make sure we'll test this scenario.

**Comment by labbati** [comment]
Targeting release 0.37.0 for this fix

**Comment by morrisonlevi** [comment]
This issue was fixed in 0.37.0. I'm not sure if you have had a chance to try again yet, @kunicmarko20. I'm going to close this ticket, but of course feel free to respond to get it re-opened if necessary.
